src/plugins/auto/auto_schedule.cpp

```diff
@@ -79,25 +79,75 @@ void AutoSchedule::GenerateWorkers(const std::string& device,
                 IdleGuard<NotBusyPriorityWorkerRequests> idleGuard{workerRequestPtr, *idleWorkerRequestsPtr};
                 workerRequestPtr->_exceptionPtr = exceptionPtr;
                 {
-                    auto capturedTask = std::move(workerRequestPtr->_task);
-                    capturedTask();
-                }
-                // try to return the request to the idle list (fails if the overall object destruction has began)
-                if (idleGuard.Release()->try_push(std::make_pair(workerRequestPtr->_index, workerRequestPtr))) {
-                    // let's try to pop a task, as we know there is at least one idle request, schedule if succeeded
-                    // if no device-agnostic tasks, let's try pop the device specific task, schedule if succeeded
-                    IE::Task t;
-                    do {
-                        _inferPipelineTasks.try_pop(t);
-                    } while (t && ScheduleToWorkerInferRequest(std::move(t)));
-                    do {
-                        _inferPipelineTasksDeviceSpecific[device]->try_pop(t);
-                    } while (t && ScheduleToWorkerInferRequest(std::move(t), device));
+                    if (workerRequestPtr->_exceptionPtr != nullptr) {
+                        if (workerRequestPtr->_testExec->count == 1) {
+                            auto capturedTask = std::move(workerRequestPtr->_task);
+                            capturedTask();
+                        } else {
+                            workerRequestPtr->_reload = true;
+                            auto capturedTask = std::move(workerRequestPtr->_testExec->_task);
+                            workerRequestPtr->_testExec->count++;
+                            selectOtherDevice(workerRequestPtr->_deviceName);
```

Comment:
what if selectOtherDevice fails, throw exception, or hang unexpectedly

Reply:
if there is no other devices in selectOtherDevice , it will retry once and end.
if load failed in selectOtherDevice, it will hang here.

Reply:
Modified the logic that notify the condition_variable and added some exception handling

Reply:
Added the lock to whole `selectOtherDevice()`, other infer request will need to wait and will get the result of previous state that show if load successfully.